File: data/recomposed_projects/deepseek-coder-33b-instruct/signature/commons-graph/evosuite-test/src/test/org/apache/commons/graph/model/BaseMutableGraph_ESTest.py
from __future__ import annotations
import time
import re
import os
import unittest
import pytest
import io
import unittest

# from src.test.org.apache.commons.graph.model.BaseMutableGraph_ESTest_scaffolding import *
from src.main.org.apache.commons.graph.model.DirectedMutableGraph import *
from src.main.org.apache.commons.graph.model.UndirectedMutableGraph import *

# from src.main.org.evosuite.runtime.EvoAssertions import *
# from src.main.org.evosuite.runtime.EvoRunnerParameters import *


class BaseMutableGraph_ESTest(unittest.TestCase):

    def test21(self) -> None:

        directedMutableGraph0 = DirectedMutableGraph()
        directedMutableGraph1 = DirectedMutableGraph()
        integer0 = 2595

        with self.assertRaises(RuntimeError):
            directedMutableGraph1.internalRemoveEdge(
                directedMutableGraph0, integer0, directedMutableGraph0
            )

    def test20(self) -> None:

        directedMutableGraph0 = DirectedMutableGraph()

        try:
            directedMutableGraph0.addEdge(None, None, None)
            self.fail("Expecting exception: RuntimeError")

        except RuntimeError as e:
            verifyException("org.apache.commons.graph.model.BaseGraph", e)

    def test19(self) -> None:

        directedMutableGraph0 = DirectedMutableGraph[int, int]()
        integer0 = int(161)

        try:
            directedMutableGraph0.addEdge(integer0, None, integer0)
            self.fail("Expecting exception: RuntimeError")

        except RuntimeError as e:
            verifyException("org.apache.commons.graph.model.BaseGraph", e)

    def test18(self) -> None:

        directed_mutable_graph0 = DirectedMutableGraph[int, int]()
        integer0 = -2

        with pytest.raises(RuntimeError):
            directed_mutable_graph0.add_edge(integer0, integer0, None)

    def test17(self) -> None:

        directedMutableGraph0 = DirectedMutableGraph[int, int]()
        integer0 = -1836
        directedMutableGraph0.addVertex(integer0)
        directedMutableGraph0.addEdge(integer0, integer0, integer0)

        # Undeclared exception in Java code, so we'll use a try-except block to catch the exception
        try:
            directedMutableGraph0.addEdge(integer0, integer0, integer0)
            self.fail("Expecting exception: RuntimeError")
        except RuntimeError as e:
            # no message in exception (getMessage() returned null)
            self.verifyException("org.apache.commons.graph.model.BaseGraph", e)

    def test16(self) -> None:

        directedMutableGraph0 = DirectedMutableGraph()

        try:
            directedMutableGraph0.addVertex(None)
            self.fail("Expecting exception: RuntimeError")

        except RuntimeError as e:
            verifyException("org.apache.commons.graph.model.BaseGraph", e)

    def test15(self) -> None:

        directedMutableGraph0 = DirectedMutableGraph[int, int]()
        integer0 = -418
        directedMutableGraph0.add_vertex(integer0)

        with pytest.raises(RuntimeError):
            directedMutableGraph0.add_vertex(integer0)

    def test14(self) -> None:

        directedMutableGraph0 = DirectedMutableGraph[int, int]()
        integer0 = -418
        directedMutableGraph0.addVertex(integer0)
        directedMutableGraph0.addEdge(integer0, integer0, integer0)
        integer1 = -418
        directedMutableGraph0.addEdge(integer1, integer1, integer1)
        self.assertEqual(1, directedMutableGraph0.getOrder())

    def test13(self) -> None:

        directedMutableGraph0 = DirectedMutableGraph()

        try:
            directedMutableGraph0.remove_edge(None)
            self.fail("Expecting exception: RuntimeError")

        except RuntimeError as e:
            verifyException("org.apache.commons.graph.model.BaseGraph", e)

    def test12(self) -> None:

        directed_mutable_graph0 = DirectedMutableGraph[int, int]()
        integer0 = int(3)

        try:
            directed_mutable_graph0.remove_edge(integer0)
            self.fail("Expecting exception: RuntimeError")

        except RuntimeError as e:
            self.verify_exception("org.apache.commons.graph.model.BaseGraph", e)

    def test11(self) -> None:

        directed_mutable_graph0 = DirectedMutableGraph()

        with pytest.raises(RuntimeError) as excinfo:
            directed_mutable_graph0.remove_vertex(None)

        assert str(excinfo.value) == "Expecting exception: RuntimeError"
        # verifyException("org.apache.commons.graph.model.BaseGraph", excinfo.value)

    def test10(self) -> None:

        directed_mutable_graph0 = DirectedMutableGraph[int, int]()
        integer0 = -418

        try:
            directed_mutable_graph0.remove_vertex(integer0)
            self.fail("Expecting exception: RuntimeError")

        except RuntimeError as e:
            self.verify_exception("org.apache.commons.graph.model.BaseGraph", e)

    def test09(self) -> None:

        directedMutableGraph0 = DirectedMutableGraph[int, int]()
        integer0 = -2142346125
        directedMutableGraph0.addVertex(integer0)
        directedMutableGraph0.decorateRemoveVertex(integer0)

        # Undeclared exception handling
        try:
            directedMutableGraph0.addEdge(integer0, integer0, integer0)
            self.fail("Expecting exception: RuntimeError")

        except RuntimeError as e:
            # no message in exception (getMessage() returned null)
            self.verifyException(
                "org.apache.commons.graph.model.DirectedMutableGraph", e
            )

    def test08(self) -> None:

        directedMutableGraph0 = DirectedMutableGraph[int, int]()
        integer0 = int(0)
        directedMutableGraph0.addVertex(integer0)

        try:
            directedMutableGraph0.internalAddEdge(integer0, None, None)
            self.fail("Expecting exception: RuntimeError")

        except RuntimeError as e:
            verifyException("org.apache.commons.graph.utils.Assertions", e)

    def test07(self) -> None:

        undirectedMutableGraph0 = UndirectedMutableGraph()

        with self.assertRaises(RuntimeError):
            undirectedMutableGraph0.internalAddEdge(None, None, None)

        verifyException("org.apache.commons.graph.model.BaseMutableGraph", RuntimeError)

    def test06(self) -> None:

        directedMutableGraph0 = DirectedMutableGraph()

        with pytest.raises(RuntimeError):
            directedMutableGraph0.internalRemoveEdge(None, None, None)

        # Impossible to construct a Vertex with a null head
        verifyException("org.apache.commons.graph.utils.Assertions", RuntimeError)

    def test05(self) -> None:

        linkedHashSet0 = set()
        directedMutableGraph0 = DirectedMutableGraph()
        directedMutableGraph0.addVertex(linkedHashSet0)
        directedMutableGraph0.internalRemoveEdge(
            linkedHashSet0, linkedHashSet0, linkedHashSet0
        )
        assert linkedHashSet0 == set()

    def test04(self) -> None:

        directedMutableGraph0 = DirectedMutableGraph[int, int]()
        integer0 = -437
        integer1 = -437

        try:
            directedMutableGraph0.add_edge(integer0, integer1, integer1)
            self.fail("Expecting exception: RuntimeError")

        except RuntimeError as e:
            self.verify_exception("org.apache.commons.graph.model.BaseGraph", e)

    def test03(self) -> None:

        directedMutableGraph0 = DirectedMutableGraph[int, int]()
        integer0 = -437
        directedMutableGraph0.addVertex(integer0)
        integer1 = -437
        directedMutableGraph0.addEdge(integer0, integer1, integer1)
        self.assertEqual(1, directedMutableGraph0.getOrder())

    def test02(self) -> None:

        directedMutableGraph0 = DirectedMutableGraph[int, int]()
        integer0 = -1836
        directedMutableGraph0.addVertex(integer0)
        integer1 = -1836
        directedMutableGraph0.addEdge(integer0, integer1, integer0)
        self.assertEqual(1, directedMutableGraph0.getSize())

    def test01(self) -> None:

        directedMutableGraph0 = DirectedMutableGraph[int, int]()
        integer0 = int(3)
        directedMutableGraph0.addVertex(integer0)
        integer1 = int(3)
        directedMutableGraph0.internalAddEdge(integer1, integer0, integer0)
        directedMutableGraph0.removeEdge(integer0)
        self.assertEqual(0, directedMutableGraph0.getSize())

    def test00(self) -> None:

        linkedHashSet0 = set()
        directedMutableGraph0 = DirectedMutableGraph()
        directedMutableGraph0.addVertex(linkedHashSet0)
        directedMutableGraph0.addEdge(linkedHashSet0, linkedHashSet0, linkedHashSet0)
        linkedHashSet1 = set()
        directedMutableGraph0.removeVertex(linkedHashSet1)
        self.assertEqual(0, len(linkedHashSet1))

File: java_projects/cleaned_final_projects_evosuite/commons-graph/src/test/java/org/apache/commons/graph/model/BaseMutableGraph_ESTest.java

/*
  Licensed to the Apache Software Foundation (ASF) under one or more
  contributor license agreements.  See the NOTICE file distributed with
  this work for additional information regarding copyright ownership.
  The ASF licenses this file to You under the Apache License, Version 2.0
  (the "License"); you may not use this file except in compliance with
  the License.  You may obtain a copy of the License at

      http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
 */


package org.apache.commons.graph.model;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import java.util.LinkedHashSet;
import org.apache.commons.graph.model.DirectedMutableGraph;
import org.apache.commons.graph.model.UndirectedMutableGraph;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true) 
public class BaseMutableGraph_ESTest extends BaseMutableGraph_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test00()  throws Throwable  {
      LinkedHashSet<Integer> linkedHashSet0 = new LinkedHashSet<Integer>();
      DirectedMutableGraph<LinkedHashSet<Integer>, LinkedHashSet<Integer>> directedMutableGraph0 = new DirectedMutableGraph<LinkedHashSet<Integer>, LinkedHashSet<Integer>>();
      directedMutableGraph0.addVertex(linkedHashSet0);
      directedMutableGraph0.addEdge(linkedHashSet0, linkedHashSet0, linkedHashSet0);
      LinkedHashSet<Integer> linkedHashSet1 = new LinkedHashSet<Integer>();
      directedMutableGraph0.removeVertex(linkedHashSet1);
      assertEquals(0, linkedHashSet1.size());
  }

  @Test(timeout = 4000)
  public void test01()  throws Throwable  {
      DirectedMutableGraph<Integer, Integer> directedMutableGraph0 = new DirectedMutableGraph<Integer, Integer>();
      Integer integer0 = new Integer(3);
      directedMutableGraph0.addVertex(integer0);
      Integer integer1 = new Integer(3);
      directedMutableGraph0.internalAddEdge(integer1, integer0, integer0);
      directedMutableGraph0.removeEdge(integer0);
      assertEquals(0, directedMutableGraph0.getSize());
  }

  @Test(timeout = 4000)
  public void test02()  throws Throwable  {
      DirectedMutableGraph<Integer, Integer> directedMutableGraph0 = new DirectedMutableGraph<Integer, Integer>();
      Integer integer0 = new Integer((-1836));
      directedMutableGraph0.addVertex(integer0);
      Integer integer1 = new Integer((-1836));
      directedMutableGraph0.addEdge(integer0, integer1, integer0);
      assertEquals(1, directedMutableGraph0.getSize());
  }

  @Test(timeout = 4000)
  public void test03()  throws Throwable  {
      DirectedMutableGraph<Integer, Integer> directedMutableGraph0 = new DirectedMutableGraph<Integer, Integer>();
      Integer integer0 = new Integer((-437));
      directedMutableGraph0.addVertex(integer0);
      Integer integer1 = new Integer((-437));
      directedMutableGraph0.addEdge(integer0, integer1, integer1);
      assertEquals(1, directedMutableGraph0.getOrder());
  }

  @Test(timeout = 4000)
  public void test04()  throws Throwable  {
      DirectedMutableGraph<Integer, Integer> directedMutableGraph0 = new DirectedMutableGraph<Integer, Integer>();
      Integer integer0 = new Integer((-437));
      Integer integer1 = new Integer((-437));
      // Undeclared exception!
      try { 
        directedMutableGraph0.addEdge(integer0, integer1, integer1);
        fail("Expecting exception: RuntimeException");
      
      } catch(RuntimeException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.apache.commons.graph.model.BaseGraph", e);
      }
  }

  @Test(timeout = 4000)
  public void test05()  throws Throwable  {
      LinkedHashSet<Integer> linkedHashSet0 = new LinkedHashSet<Integer>();
      DirectedMutableGraph<LinkedHashSet<Integer>, LinkedHashSet<Integer>> directedMutableGraph0 = new DirectedMutableGraph<LinkedHashSet<Integer>, LinkedHashSet<Integer>>();
      directedMutableGraph0.addVertex(linkedHashSet0);
      directedMutableGraph0.internalRemoveEdge(linkedHashSet0, linkedHashSet0, linkedHashSet0);
      assertTrue(linkedHashSet0.isEmpty());
  }

  @Test(timeout = 4000)
  public void test06()  throws Throwable  {
      DirectedMutableGraph<Integer, Integer> directedMutableGraph0 = new DirectedMutableGraph<Integer, Integer>();
      // Undeclared exception!
      try { 
        directedMutableGraph0.internalRemoveEdge((Integer) null, (Integer) null, (Integer) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // Impossible to construct a Vertex with a null head
         //
         verifyException("org.apache.commons.graph.utils.Assertions", e);
      }
  }

  @Test(timeout = 4000)
  public void test07()  throws Throwable  {
      UndirectedMutableGraph<Integer, Integer> undirectedMutableGraph0 = new UndirectedMutableGraph<Integer, Integer>();
      // Undeclared exception!
      try { 
        undirectedMutableGraph0.internalAddEdge((Integer) null, (Integer) null, (Integer) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.apache.commons.graph.model.BaseMutableGraph", e);
      }
  }

  @Test(timeout = 4000)
  public void test08()  throws Throwable  {
      DirectedMutableGraph<Integer, Integer> directedMutableGraph0 = new DirectedMutableGraph<Integer, Integer>();
      Integer integer0 = new Integer(0);
      directedMutableGraph0.addVertex(integer0);
      // Undeclared exception!
      try { 
        directedMutableGraph0.internalAddEdge(integer0, (Integer) null, (Integer) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // Impossible to construct a Vertex with a null tail
         //
         verifyException("org.apache.commons.graph.utils.Assertions", e);
      }
  }

  @Test(timeout = 4000)
  public void test09()  throws Throwable  {
      DirectedMutableGraph<Integer, Integer> directedMutableGraph0 = new DirectedMutableGraph<Integer, Integer>();
      Integer integer0 = new Integer((-2142346125));
      directedMutableGraph0.addVertex(integer0);
      directedMutableGraph0.decorateRemoveVertex(integer0);
      // Undeclared exception!
      try { 
        directedMutableGraph0.addEdge(integer0, integer0, integer0);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.apache.commons.graph.model.DirectedMutableGraph", e);
      }
  }

  @Test(timeout = 4000)
  public void test10()  throws Throwable  {
      DirectedMutableGraph<Integer, Integer> directedMutableGraph0 = new DirectedMutableGraph<Integer, Integer>();
      Integer integer0 = new Integer((-418));
      // Undeclared exception!
      try { 
        directedMutableGraph0.removeVertex(integer0);
        fail("Expecting exception: RuntimeException");
      
      } catch(RuntimeException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.apache.commons.graph.model.BaseGraph", e);
      }
  }

  @Test(timeout = 4000)
  public void test11()  throws Throwable  {
      DirectedMutableGraph<Integer, Integer> directedMutableGraph0 = new DirectedMutableGraph<Integer, Integer>();
      // Undeclared exception!
      try { 
        directedMutableGraph0.removeVertex((Integer) null);
        fail("Expecting exception: RuntimeException");
      
      } catch(RuntimeException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.apache.commons.graph.model.BaseGraph", e);
      }
  }

  @Test(timeout = 4000)
  public void test12()  throws Throwable  {
      DirectedMutableGraph<Integer, Integer> directedMutableGraph0 = new DirectedMutableGraph<Integer, Integer>();
      Integer integer0 = new Integer(3);
      // Undeclared exception!
      try { 
        directedMutableGraph0.removeEdge(integer0);
        fail("Expecting exception: RuntimeException");
      
      } catch(RuntimeException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.apache.commons.graph.model.BaseGraph", e);
      }
  }

  @Test(timeout = 4000)
  public void test13()  throws Throwable  {
      DirectedMutableGraph<Object, Integer> directedMutableGraph0 = new DirectedMutableGraph<Object, Integer>();
      // Undeclared exception!
      try { 
        directedMutableGraph0.removeEdge((Integer) null);
        fail("Expecting exception: RuntimeException");
      
      } catch(RuntimeException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.apache.commons.graph.model.BaseGraph", e);
      }
  }

  // @Test(timeout = 4000)
  // public void test14()  throws Throwable  {
  //     DirectedMutableGraph<Integer, Integer> directedMutableGraph0 = new DirectedMutableGraph<Integer, Integer>();
  //     Integer integer0 = new Integer((-418));
  //     directedMutableGraph0.addVertex(integer0);
  //     directedMutableGraph0.addEdge(integer0, integer0, integer0);
  //     Integer integer1 = new Integer((-418));
  //     directedMutableGraph0.addEdge(integer1, integer1, integer1);
  //     assertEquals(1, directedMutableGraph0.getOrder());
  // }

  @Test(timeout = 4000)
  public void test15()  throws Throwable  {
      DirectedMutableGraph<Integer, Integer> directedMutableGraph0 = new DirectedMutableGraph<Integer, Integer>();
      Integer integer0 = new Integer((-418));
      directedMutableGraph0.addVertex(integer0);
      // Undeclared exception!
      try { 
        directedMutableGraph0.addVertex(integer0);
        fail("Expecting exception: RuntimeException");
      
      } catch(RuntimeException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.apache.commons.graph.model.BaseGraph", e);
      }
  }

  @Test(timeout = 4000)
  public void test16()  throws Throwable  {
      DirectedMutableGraph<Integer, Integer> directedMutableGraph0 = new DirectedMutableGraph<Integer, Integer>();
      // Undeclared exception!
      try { 
        directedMutableGraph0.addVertex((Integer) null);
        fail("Expecting exception: RuntimeException");
      
      } catch(RuntimeException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.apache.commons.graph.model.BaseGraph", e);
      }
  }

  @Test(timeout = 4000)
  public void test17()  throws Throwable  {
      DirectedMutableGraph<Integer, Integer> directedMutableGraph0 = new DirectedMutableGraph<Integer, Integer>();
      Integer integer0 = new Integer((-1836));
      directedMutableGraph0.addVertex(integer0);
      directedMutableGraph0.addEdge(integer0, integer0, integer0);
      // Undeclared exception!
      try { 
        directedMutableGraph0.addEdge(integer0, integer0, integer0);
        fail("Expecting exception: RuntimeException");
      
      } catch(RuntimeException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.apache.commons.graph.model.BaseGraph", e);
      }
  }

  @Test(timeout = 4000)
  public void test18()  throws Throwable  {
      DirectedMutableGraph<Integer, Integer> directedMutableGraph0 = new DirectedMutableGraph<Integer, Integer>();
      Integer integer0 = new Integer((-2));
      // Undeclared exception!
      try { 
        directedMutableGraph0.addEdge(integer0, integer0, (Integer) null);
        fail("Expecting exception: RuntimeException");
      
      } catch(RuntimeException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.apache.commons.graph.model.BaseGraph", e);
      }
  }

  @Test(timeout = 4000)
  public void test19()  throws Throwable  {
      DirectedMutableGraph<Integer, Integer> directedMutableGraph0 = new DirectedMutableGraph<Integer, Integer>();
      Integer integer0 = new Integer(161);
      // Undeclared exception!
      try { 
        directedMutableGraph0.addEdge(integer0, (Integer) null, integer0);
        fail("Expecting exception: RuntimeException");
      
      } catch(RuntimeException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.apache.commons.graph.model.BaseGraph", e);
      }
  }

  @Test(timeout = 4000)
  public void test20()  throws Throwable  {
      DirectedMutableGraph<Integer, Integer> directedMutableGraph0 = new DirectedMutableGraph<Integer, Integer>();
      // Undeclared exception!
      try { 
        directedMutableGraph0.addEdge((Integer) null, (Integer) null, (Integer) null);
        fail("Expecting exception: RuntimeException");
      
      } catch(RuntimeException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.apache.commons.graph.model.BaseGraph", e);
      }
  }

  @Test(timeout = 4000)
  public void test21()  throws Throwable  {
      DirectedMutableGraph<Integer, Integer> directedMutableGraph0 = new DirectedMutableGraph<Integer, Integer>();
      DirectedMutableGraph<Object, Integer> directedMutableGraph1 = new DirectedMutableGraph<Object, Integer>();
      Integer integer0 = new Integer(2595);
      // Undeclared exception!
      try { 
        directedMutableGraph1.internalRemoveEdge(directedMutableGraph0, integer0, directedMutableGraph0);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.apache.commons.graph.model.BaseMutableGraph", e);
      }
  }
}
